### PR TITLE
update upstream cni to 1.5.1

### PR DESCRIPTION
**Pull request by haouc**
**What type of PR is this?**

**Which issue does this PR fix?**:

**What does this PR do / Why do we need it?**:
This is upgrading upstream CNI binary to v1.5.1.

**Testing done on this change**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades? Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -279,7 +279,7 @@ docker-metrics-test:     ## Run metrics helper unit test suite in a container.
 		make metrics-unit-test
 
 # Fetch the CNI plugins
-plugins: FETCH_VERSION=1.4.0
```
Comment (orsenthil) [inline Makefile:282]: We have to look at the change log to see if these introduce any backwards incompatiblity changes. We had revert it back to 1.4.x sometime ago, if I remember correctly. 

cc: @M00nF1sh  do you recollect?

**Comment by haouc** [comment]
Diff
https://github.com/containernetworking/plugins/compare/v1.4.0...v1.5.1

File: Makefile (hunk 1 above)
Comment (haouc) [reply Makefile:282]: Diff
https://github.com/containernetworking/plugins/compare/v1.4.0...v1.5.1

Comment (orsenthil) [reply Makefile:282]: This update is fine. We aren't changing the CNI spec here.

**Comment by orsenthil** [comment]
I don't see any backwards incompatible changes listed in the releases
- https://github.com/containernetworking/plugins/releases
- Most of this diff is https://github.com/containernetworking/plugins/compare/v1.4.0...v1.5.1 is bug fixes and tests.

So, this upgrade looks good to me.

**Review by orsenthil — APPROVED** [review]

**Pull request merged**